### PR TITLE
RFC: LLM -> StrInStrOutLLM

### PR DESCRIPTION
**Pull request by dev2049**
Find this interface name very confusing (not specific enough). Thoughts on updating? Open to other names, just want it to be specific.

Think it's backwards compat but very well might be overlooking something

### CODE REVIEW

**Comment by dev2049** [comment]
what's opposite of "batched"? non-batched? could also call it something like that (NonBatchedLLM)

File: langchain/llms/base.py [hunk 1]
```diff
@@ -365,30 +365,52 @@ def save(self, file_path: Union[Path, str]) -> None:
             raise ValueError(f"{save_path} must be json or yaml")
 
 
-class LLM(BaseLLM):
+class StrInStrOutLLM(BaseLLM):
```
Comment (dev2049) [inline langchain/llms/base.py:368]: main file

**Comment by vowelparrot** [comment]
Hm I agree the `_call` naming is confusing and it'd be easier if we went the other way (generate -> batch_generate). Could call it something like `_call_single()` or something

**Comment by dev2049** [comment]
> Hm I agree the `_call` naming is confusing and it'd be easier if we went the other way (generate -> batch_generate). Could call it something like `_call_single()` or something

i like idea of this being generate and other one being batch_generate, just worried that that may be too big of a change at this pt

i like generate more than call, call doesn't mean much to me. i also think its more than just "single", which to me would imply you're returning a single Generate object. but this is even less than that, its a single string output

**Comment by dev2049** [comment]
another idea: SimpleLLM

**Pull request closed without merging**